com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v3/ApplicationsCloudModelBuilder.java

```diff
@@ -25,69 +31,90 @@
 import com.sap.cloud.lm.sl.mta.util.PropertiesUtil;
 
 public class ApplicationsCloudModelBuilder extends com.sap.cloud.lm.sl.cf.core.cf.v2.ApplicationsCloudModelBuilder {
-    
+
     private static final int MTA_MAJOR_VERSION = 3;
 
     public ApplicationsCloudModelBuilder(DeploymentDescriptor deploymentDescriptor, CloudModelConfiguration configuration,
         DeployedMta deployedMta, SystemParameters systemParameters, XsPlaceholderResolver xsPlaceholderResolver, String deployId) {
         super(deploymentDescriptor, configuration, deployedMta, systemParameters, xsPlaceholderResolver, deployId);
     }
-    
+
     public ApplicationsCloudModelBuilder(DeploymentDescriptor deploymentDescriptor, CloudModelConfiguration configuration,
         DeployedMta deployedMta, SystemParameters systemParameters, XsPlaceholderResolver xsPlaceholderResolver, String deployId,
         UserMessageLogger userMessageLogger) {
         super(deploymentDescriptor, configuration, deployedMta, systemParameters, xsPlaceholderResolver, deployId, userMessageLogger);
     }
-    
+
+    @Override
+    public DeploymentMode getDeploymentMode() {
+        DeploymentDescriptor descriptorV3 = cast(deploymentDescriptor);
+        boolean parallelDeploymentsEnabled = (Boolean) descriptorV3.getParameters()
+            .getOrDefault(SupportedParameters.ENABLE_PARALLEL_DEPLOYMENTS, false);
+        return parallelDeploymentsEnabled ? DeploymentMode.PARALLEL : DeploymentMode.SEQUENTIAL;
+    }
+
     @Override
     protected HandlerFactory createHandlerFactory() {
         return new HandlerFactory(MTA_MAJOR_VERSION);
     }
-    
+
+    @Override
+    protected CloudApplicationExtended getApplication(com.sap.cloud.lm.sl.mta.model.v1.Module module) {
+        CloudApplicationExtended app = super.getApplication(module);
+        List<String> deployedAfter = emptyIfNull(((Module) module).getDeployedAfter());
```

Comment:
Here you're converting a set to a list. Use SetUtils.emptyIfNull instead of ListUtils.emptyIfNull. And don't static import it.

Btw, [??](http://s3-eu-west-1.amazonaws.com/malikafavre/project_large_images/69_image1.png?mtime=1433245318)'th comment. :D

Reply:
I could use SetUtils.emptyIfNull if getDeployedAfter returned a set, but that is not the case, therefore the usage of ListUtils.emptyIfNull. And the conversion to set should happen after the value is ensured not to be null.